modules/045-snapshot-controller/templates/snapshot-validation-webhook/deployment.yaml

```diff
@@ -0,0 +1,77 @@
+{{- if (.Values.global.enabledModules | has "vertical-pod-autoscaler-crd") }}
+---
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: d8-snapshot-validation-webhook
+  namespace: kube-system
+  {{- include "helm_lib_module_labels" (list . (dict "app" "d8-snapshot-validation-webhook" "workload-resource-policy.deckhouse.io" "master")) | nindent 2 }}
+spec:
+  targetRef:
+    apiVersion: "apps/v1"
+    kind: Deployment
+    name: d8-snapshot-validation-webhook
+  updatePolicy:
+    updateMode: "Auto"
+{{- end }}
+---
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  annotations:
+    helm.sh/hook: post-upgrade, post-install
+    helm.sh/hook-delete-policy: before-hook-creation
+  name: d8-snapshot-validation-webhook
+  namespace: kube-system
+  {{- include "helm_lib_module_labels" (list . (dict "app" "d8-snapshot-validation-webhook" )) | nindent 2 }}
+spec:
+  minAvailable: {{ include "helm_lib_is_ha_to_value" (list . 1 0) }}
+  selector:
+    matchLabels:
+      app: d8-snapshot-validation-webhook
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: d8-snapshot-validation-webhook
+  namespace: kube-system
```

Comment:
Do not like deploying this to the kube-system namespaces. Is there something which presents us to deploy it to a dedicated namespace?

Reply:
Actually not, I was inspired by other cluster-wide services, eg: `d8-kube-dns` `d8-kube-proxy` and `d8-control-plane-manager`. `snapshot-controller` is extending Kubernetes API by adding general APIs for all CSI drivers in cluster. Should I move it?

Reply:
We also have extensions like bashibile-apiserver, and authorization webhook.